src/Adapter/Presenter/Product/ProductLazyArray.php

```diff
@@ -284,6 +290,23 @@ public function getFileSizeFormatted()
         return null;
     }
 
+    /**
+     * @arrayAccess
+     * @return ProductAttachmentLazyArray[]
+     * @throws \ReflectionException
+     */
+    public function getAttachments()
+    {
+        if (null === $this->attachments) {
+            $this->attachments = array();
+            foreach ($this->product['attachments'] as $attachment) {
+                $this->attachments[] = new ProductAttachmentLazyArray($attachment);
```

Comment:
I was not sure, it might be over engineered but I wanted to avoid instanciating the adapters for nothing, you think it should be initialized in the constructor?